src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -68,30 +76,18 @@ public boolean isRunning(){
         }
     }
     public void Stop(){
-        if(mBTConnector != null){
-            mBTConnector.Stop();
-            mBTConnector = null;
-        }
-
-        if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                }
-            }
 
-            mServerSocketList.clear();
+        BTConnector tmpCon = mBTConnector;
+        mBTConnector = null;
+        if(tmpCon != null){
+            tmpCon.Stop();// this is synchronized function
         }
 
-        if(mBtToRequestSocket != null) {
-            print_debug("Disconnect:::Stop : mBtToRequestSocket");
-            mBtToRequestSocket.Stop();
-            mBtToRequestSocket = null;
-        }
+        //disconnect all incoming connections
+        DisconnectIncomingConnections();
+
+        // disconnect outgoing connection
+        Disconnect("");
     }
 
     public boolean Disconnect(String peerId){
```

Comment:
@DrJukka This method really confuses me. I thought that you said that BlueTooth supports connecting to multiple different devices simultaneously. But this Disconnect method seems to assume that we will only connect to a single peer at a time. What am I missing?